shared/observability/src/airflow_shared/observability/metrics/statsd_logger.py

```diff
@@ -155,13 +158,25 @@ def timer(
         return Timer()
 
 
-def get_statsd_logger(cls) -> SafeStatsdLogger:
+def get_statsd_logger(
+    cls,
+    stats_class: Callable[[str], str] | None = None,
+    host: str | None = None,
+    port: int | None = None,
+    prefix: str | None = None,
+    ipv6: bool = False,
+    influxdb_tags_enabled: bool = False,
+    statsd_disabled_tags: str | None = None,
+    metrics_allow_list: str | None = None,
+    metrics_block_list: str | None = None,
+    stat_name_handler: Callable[[str], str] | None = None,
+    statsd_influxdb_enabled: bool = False,
+) -> SafeStatsdLogger:
     """Return logger for StatsD."""
     # no need to check for the scheduler/statsd_on -> this method is only called when it is set
     # and previously it would crash with None is callable if it was called without it.
     from statsd import StatsClient
 
-    stats_class = conf.getimport("metrics", "statsd_custom_client_path", fallback=None)
     if stats_class:
         if not issubclass(stats_class, StatsClient):
             raise AirflowConfigException(
```

Comment:
Wait since we inject this and dont really read `conf`, we can just get rid of this check or move it factory where we actually read `conf`?

Reply:
`conf.getimport("metrics", "statsd_custom_client_path", fallback=None)` I mean